### PR TITLE
adds ci specific envs for reqProc booting

### PR DESCRIPTION
**Pull request by scriptnull**
https://github.com/Shippable/node/issues/148

Build with docker push on docker version 17.06 is happening after initializing the node with this script.
image

### CODE REVIEW

File: initScripts/x86_64__Ubuntu_16.04__Docker_17.06.sh [hunk 1]
```diff
@@ -235,6 +237,21 @@ setup_mounts() {
...
+  if [ -z "$installed_docker_version" ]; then
+    installed_docker_version=$(sudo docker version --format {{.Server.Version}})
+  fi
+
+  if [ -f "$LEGACY_CI_DOCKER_CLIENT_LATEST" ]; then
+    is_docker_legacy=false
+    docker_client_location=$LEGACY_CI_DOCKER_CLIENT_LATEST
+  fi
+
+  __process_msg "Docker client location on host: $docker_client_location"
```
Comment (scriptnull) [inline initScripts/x86_64__Ubuntu_16.04__Docker_17.06.sh]: this script uses `echo` for printing most stuff. `process_msg` is used here and in `pull_cexec`. Should I be using `echo` in these places to maintain uniformity ?

File: initScripts/x86_64__Ubuntu_16.04__Docker_17.06.sh [hunk 2]
```diff
@@ -35,6 +35,8 @@ export LEGACY_CI_CEXEC_LOCATION_ON_HOST="/home/shippable/cexec"
 export DEFAULT_TASK_CONTAINER_MOUNTS="-v $BUILD_DIR:$BUILD_DIR \
   -v $REQEXEC_DIR:/reqExec"
 export DEFAULT_TASK_CONTAINER_OPTIONS="-d --rm"
+export LEGACY_CI_DOCKER_CLIENT_LEGACY="/usr/bin/docker"
```
Comment (shrivara) [inline initScripts/x86_64__Ubuntu_16.04__Docker_17.06.sh]: We can remove legacy and assume that this is the latest. We're installing Docker 17.06 here.

File: initScripts/x86_64__Ubuntu_16.04__Docker_17.06.sh (hunk 1 above)
Comment (ric03uec) [reply initScripts/x86_64__Ubuntu_16.04__Docker_17.06.sh]: ya. lets change it to echo for now. we'll refactor this to use process_msg globally before releasing

File: initScripts/x86_64__Ubuntu_16.04__Docker_17.06.sh (hunk 2 above)
Comment (ric03uec) [reply initScripts/x86_64__Ubuntu_16.04__Docker_17.06.sh]: @shrivara wont' we need this if old users start running builds on this platform. im talking about running legacy ci running on dynamic nodes?

Comment (shrivara) [reply initScripts/x86_64__Ubuntu_16.04__Docker_17.06.sh]: Yes, but we've moved this out from boot.sh to here. So, we'll know exactly which version of Docker we're installing.

Comment (ric03uec) [reply initScripts/x86_64__Ubuntu_16.04__Docker_17.06.sh]: ah. ok. so the point here is that these two variables should only be exported based on which version of docker we're running. if its legacy then only export legacy stuff else import latest stuff right?

Comment (shrivara) [reply initScripts/x86_64__Ubuntu_16.04__Docker_17.06.sh]: Yes

**Pull request merged**